data/transactions/logic/README.md

```diff
@@ -86,35 +217,39 @@ An application transaction must indicate the action to be taken following the ex
 
 This summary is supplemented by more detail in the [opcodes document](TEAL_opcodes.md).
 
-Some operations 'panic' and immediately fail the program.
-A transaction checked by a program that panics is not valid.
-A contract account governed by a buggy program might not have a way to get assets back out of it. Code carefully.
+Some operations immediately fail the program.
+A transaction checked by a program that fails is not valid.
+An account governed by a buggy program might not have a way to get assets back out of it. Code carefully.
 
-### Arithmetic, Logic, and Cryptographic Operations
+In the documentation for each opcode, the stack arguments that are
+popped are referred to alphabetically, beginning with the deepest
+argument as `A`.  These arguments are shown in the opcode description,
+and if the opcode must be of a specific type, it is noted there.  All
+opcodes fail if a specified type is incorrect.
 
-For one-argument ops, `X` is the last element on the stack, which is typically replaced by a new value.
+If an opcode pushes more than one result, the values are named for
+ease of exposition and clarity concerning their stack positions.  When
+an opcode manipulates the stack in such a way that a value changes
+position but is otherwise unchanged, the name of the output on the
+return stack matches the name of the input value.
 
-For two-argument ops, `A` is the penultimate element on the stack and `B` is the top of the stack. These typically result in popping A and B from the stack and pushing the result.
-
-For three-argument ops, `A` is the element two below the top, `B` is the penultimate stack element and `C` is the top of the stack. These operations typically pop A, B, and C from the stack and push the result.
+### Arithmetic, Logic, and Cryptographic Operations
 
-| Op | Description |
-| --- | --- |
-| `sha256` | SHA256 hash of value X, yields [32]byte |
-| `keccak256` | Keccak256 hash of value X, yields [32]byte |
-| `sha512_256` | SHA512_256 hash of value X, yields [32]byte |
+| Opcode | Description |
+| - | -- |
+| `sha256` | SHA256 hash of value A, yields [32]byte |
+| `keccak256` | Keccak256 hash of value A, yields [32]byte |
+| `sha512_256` | SHA512_256 hash of value A, yields [32]byte |
 | `ed25519verify` | for (data A, signature B, pubkey C) verify the signature of ("ProgData" \|\| program_hash \|\| data) against the pubkey => {0 or 1} |
 | `ecdsa_verify v` | for (data A, signature B, C and pubkey D, E) verify the signature of the data against the pubkey => {0 or 1} |
-| `ecdsa_pk_recover v` | for (data A, recovery id B, signature C, D) recover a public key => [*... stack*, X, Y] |
-| `ecdsa_pk_decompress v` | decompress pubkey A into components X, Y => [*... stack*, X, Y] |
+| `ecdsa_pk_recover v` | for (data A, recovery id B, signature C, D) recover a public key |
```

Comment:
I've made it so that stack changes are automatically documented so I removed the extra notation in some of the documentation strings.  For example, ecdsa_recover now says this in it's header:
- Stack: ..., A: []byte, B: uint64, C: []byte, D: []byte => ..., X: []byte, Y: []byte

I think that covers what you intended. Is that right?